docs/usage/modules/versioning.md

```diff
@@ -4,11 +4,29 @@ Once Managers have extracted dependencies, and Datasources have located availabl
 The "versioning" is different for each package manager, because different package managers use different versioning schemes.
 For example, `npm` uses`1.0.0-beta.1` and `pip` uses `1.0.0b1`.
 
-## Configuring Versioning
+## Why you might need to tinker with the versioning
+
+Renovate tries to do the right thing out-of-the-box most of the time.
+It's impossible to support **all** versioning schemes, so sometimes you need to tell the bot what versioning scheme it should use.
 
 You can manually configure/override the `versioning` value for a particular dependency.
 You generally won't need to override the defaults for ecosystems which enforce a strict version scheme like `npm`.
-Configuring or overriding the default `versioning` can be helpful for ecosystems like Docker, where versioning is barely a "convention". e.g.
+
+Configuring or overriding the default `versioning` can be helpful for ecosystems like Docker or helm-values, where versioning is barely a "convention".
+
+## General concepts behind overriding versioning
+
+TODO:
+
+- Explain how/why to use `packageRules`
+- Explain what kind of things you can set with `packageRules`
+- Explain what are the most common proporties you will need in `packageRules` to get what you want
+
+## Examples of commonly needed versioning overrides
+
```

Comment:
```suggestion
## Examples of versioning overrides

### Overriding Docker versioning to use a versioning specific for a package

```json
{
  "packageRules": [
    {
      "matchDatasources": ["docker"],
      "matchPackageNames": ["python"],
      "versionScheme": "pep440"
    }
  ]
}
```

### Using a custom regex versioning scheme

```json
{
  "packageRules": [
    {
      "matchPackageNames": ["foo/bar"],
      "versionScheme": "regex:^(?<compatibility>.*)-v?(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<patch>\\d+)?$"
    }
  ]
}
```

Reply:
I've made the changes you wanted here, and added the `h2` for Using a custom regex versioning scheme as you suggested.
Resolving this conversation. 😄